cmd/crates/soroban-test/tests/it/integration/bindings.rs

```diff
@@ -0,0 +1,35 @@
+use soroban_test::{TestEnv, LOCAL_NETWORK_PASSPHRASE};
+use super::util::deploy_swap;
+
+pub const OUTPUT_DIR: &str = "./bindings-output";
+
+
+#[tokio::test]
+async fn invoke_test_generate_typescript_bindings() {
+    let sandbox = &TestEnv::new();
+    let contract_id = deploy_swap(sandbox).await;
+    let cmd = sandbox.cmd_arr::<soroban_cli::commands::contract::bindings::typescript::Cmd>(&[
+        "--network-passphrase",
+        LOCAL_NETWORK_PASSPHRASE,
+        "--rpc-url",
+        &sandbox.rpc_url,
+        "--output-dir",
+        OUTPUT_DIR,
+        "--overwrite",
+        "--contract-id",
+        &contract_id.to_string(),
+    ]);
+
+    let result = sandbox.run_cmd_with(cmd, "test").await;
+
+    assert!(result.is_ok(), "Failed to generate TypeScript bindings");
+
+    let output_dir = std::path::Path::new(OUTPUT_DIR);
+    assert!(output_dir.exists(), "Output directory does not exist");
+
+    let files = std::fs::read_dir(output_dir).expect("Failed to read output directory");
+    assert!(files.count() > 0, "No files generated in the output directory");
+
+    // Clean up: remove the output directory and its contents
+    std::fs::remove_dir_all(OUTPUT_DIR).expect("Failed to delete output directory");
```

Comment:
```suggestion
```
`TestEnv` uses temp_dir, which when dropped deletes the folder! This can be an issue if it's dropped early. This is why having an interface that lets you pass a function, which has the `&TestEnv` as a argument is preferred since it can't be dropped by the test.

It's currently hard to do with async functions though or at least last time I tried.

Reply:
fixed